packages/ui/src/components/input.tsx

```diff
@@ -1,19 +1,61 @@
-import * as React from 'react'
+import { forwardRef, InputHTMLAttributes } from 'react'
+
+import { cva, VariantProps } from 'class-variance-authority'
 
 import { cn } from '../utils/cn'
+import { Text } from './'
 
-export interface BaseInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
-  isExtended?: boolean
+export interface BaseInputProps
+  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'size'>,
+    VariantProps<typeof inputVariants> {
+  wrapperClassName?: string
+  error?: string
 }
 
-const BaseInput = React.forwardRef<HTMLInputElement, InputProps>(({ className, type, isExtended, ...props }, ref) => {
-  const commonClassName = 'bg-transparent text-foreground-1 px-2.5 py-1'
-  const specificClassNames = isExtended
-    ? 'border-none grow focus-visible:outline-none'
-    : 'flex h-8 w-full rounded border border-border-2 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-foreground-4 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50'
+const inputVariants = cva(
+  'remove-autocomplete-styles bg-transparent placeholder:text-foreground-5 text-foreground-1 px-3 py-1',
+  {
+    variants: {
+      variant: {
+        default:
+          'flex h-9 w-full rounded border text-sm shadow-sm transition-colors file:border-0 focus-visible:outline-none file:bg-transparent file:text-sm file:font-medium placeholder:text-foreground-4 disabled:cursor-not-allowed disabled:opacity-50',
+        extended: 'grow border-none focus-visible:outline-none'
+      },
+      theme: {
+        default: 'border-borders-2 focus-visible:border-borders-3',
+        danger: 'border-borders-danger'
+      }
+    }
+  }
+)
 
-  return <input className={cn(commonClassName, specificClassNames, className)} type={type} ref={ref} {...props} />
-})
+const BaseInput = forwardRef<HTMLInputElement, BaseInputProps>(
+  ({ className, type, variant = 'default', theme = 'default', error, wrapperClassName, ...props }, ref) => {
+    return (
+      <div className={cn('relative flex flex-col', wrapperClassName)}>
+        <input
+          className={cn(
+            inputVariants({ variant, theme }),
+            className,
+            error ? 'border-borders-danger' : 'border-borders-2'
+          )}
+          type={type}
+          ref={ref}
+          {...props}
+        />
+        {error && (
+          <Text
+            className="absolute top-full translate-y-1 text-foreground-danger leading-none tracking-tight"
```

Comment:
We already have the tracking-tight class applied to the body, so there's no need to include it here.